pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -16,6 +16,68 @@ final class Event {
       : eventName = DashEvent.analyticsCollectionEnabled,
         eventData = {'status': status};
 
+  /// Event that is emitted when a Dart or Flutter CLI command has been
+  /// executed.
+  ///
+  /// [name] - the name of the command that was executed
+  ///
+  /// [flags] - the set of names for flags and options provided to the command.
+  ///    Does not include values provided by users for options.
+  ///
+  /// [enabledExperiments] - the set of Dart language experiments enabled when
+  ///    running the command.
+  ///
+  /// [exitCode] - the process exit code set as a result of running the command.
+  Event.cliCommandExecuted({
+    required String name,
+    required List<String> flags,
+    required List<String> enabledExperiments,
```

Comment:
The types for each of these constructors should be `String`, `int`, `double`, or `bool`

Additionally, I'm a little concerned with having a list that we convert to a string because we have a 100 char limit for each of these parameters or else they will get truncated.  Do you have an idea of how much data could possibly get sent for this one?

Reply:
Done.

After talking with mit@, I've removed the `flags` parameter as it wasn't providing us much value anyway. I think we should be safe to include the set of experiments as there should only ever be a handful enabled at a given time (the full list of experiments can be found [here](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/dart/analysis/experiments.g.dart#L54), but many of them are no longer relevant).